### ISSUE TITLE
"Unexpected token :" for JSON-LD (ld+json). 

### ISSUE THREAD

**Issue by jtheletter** [eslint/eslint#9981]
If I place a JSON-LD script element in an HTML file, ESLint throws an error of "Unexpected token :". 

Google doc: https://developers.google.com/search/docs/guides/intro-structured-data

        <script type="application/ld+json">
            {
                "@context": "http://schema.org",
                "@type": "Organization",
                "url": "http://www.example.com",
                "name": "Unlimited Ball Bearings Corp.",
                "contactPoint": {
                    "@type": "ContactPoint",
                    "telephone": "+1-401-555-1212",
                    "contactType": "Customer service"
                }
            }
        </script>

Disabling ESLint in Sublime Text menubar's "Toggle" command removes the error. 

Adding `/* eslint-disable unexpected-token */` doesn't remove the error. 

Should this be reported here or at [SublimeLinter-eslint](https://github.com/SublimeLinter/SublimeLinter-eslint/issues)? 

Can this be prevented, please? 

- ESLint 4.17.0
- SublimeLinter 3.10.10
- Sublime Text 3.0
- macOS 10.13.3 High Sierra

**Comment by ilyavolodin** [comment]
By default, ESLint does not know how to lint HTML files at all. This is most likely coming from a plugin. ESLint also doesn't link JSON either. Could you please fill out provided template so we could point you to the right plugin?

**Comment by jaepage** [comment]
SublimeLinter settings: 
```
{
    "user": {
        "debug": false,
        "delay": 0,
        "error_color": "e00",
        "gutter_theme": "Packages/SublimeLinter/gutter-themes/Default/Default.gutter-theme",
        "gutter_theme_excludes": [],
        "lint_mode": "background",
        "linters": {
            "csslint": {
                "@disable": false,
                "args": [],
                "errors": "",
                "excludes": [],
                "ignore": [
                    "box-model",
                    "box-sizing",
                    "bulletproof-font-face"
                ],
                "ignore_match": [
                    "Unknown property 'unicode-range'"
                ],
                "warnings": ""
            },
            "eslint": {
                "@disable": false,
                "args": [],
                "config_filename": ".eslintrc",
                "excludes": []
            },
            "htmlhint": {
                "@disable": false,
                "args": [],
                "excludes": [],
                "ignore": []
            },
            "jshint": {
                "@disable": false,
                "args": [],
                "excludes": [
                    "*.jsx"
                ]
            },
            "json": {
                "@disable": false,
                "args": [],
                "excludes": [],
                "strict": true
            },
            "php": {
                "@disable": false,
                "args": [],
                "excludes": []
            },
            "scss": {
                "@disable": false,
                "args": [],
                "exclude-linter": "",
                "excludes": [
                    "*.css"
                ],
                "include-linter": ""
            }
        },
        "mark_style": "stippled underline",
        "no_column_highlights_line": false,
        "passive_warnings": false,
        "paths": {
            "linux": [],
            "osx": [],
            "windows": []
        },
        "python_paths": {
            "linux": [],
            "osx": [],
            "windows": []
        },
        "rc_search_limit": 3,
        "shell_timeout": 10,
        "show_errors_on_save": false,
        "show_marks_in_minimap": true,
        "syntax_map": {
            "html (django)": "html",
            "html (rails)": "html",
            "html 5": "html",
            "javascript (babel)": "javascript",
            "magicpython": "python",
            "php": "html",
            "python django": "python",
            "pythonimproved": "python"
        },
        "tooltip_fontsize": "1rem",
        "tooltip_theme": "Packages/SublimeLinter/tooltip-themes/Default/Default.tooltip-theme",
        "tooltip_theme_excludes": [],
        "tooltips": false,
        "use_current_shell_path": false,
        "warning_color": "bb0",
        "wrap_find": true
    }
}
```

**Comment by jaepage** [comment]
Are you asking my ESLint config? 
```
{
    "parserOptions": {
        "ecmaVersion": 6,
        "sourceType": "module",
        "ecmaFeatures": {
            "jsx": true
        }
    }
}
```

**Comment by ilyavolodin** [comment]
ESLint config. However, config that you pasted is not complete. With the config above, ESLint would not parse or even touch html files at all. You must have some plugins kicking configured somewhere.

**Comment by jaepage** [comment]
The error disappears when I disable ESLint, reappears when I re-enable. The error persists when I disable all other linters.

**Comment by ilyavolodin** [comment]
Yes, if you disable ESLint, none of your eslint plugins will execute. So that's expected. ESLint is purely a JavaScript linter. It doesn't know (and doesn't try) to lint HTML files or JSON. In order to lint those (and get any errors at all), you have to have some plugins enabled somewhere in your .eslintrc.* files.

**Comment by jaepage** [comment]
That's my only `.eslintrc` file. Could you try inserting the above JSON-LD sample script from Google’s dev docs into an HTML file to see if it shows differently for you?

**Comment by jaepage** [comment]
Also, `npm list --depth=0` shows `└── (empty)`, and `npm list -g --depth=0` shows: 

```
/usr/local/lib
├── babel@6.23.0
├── babel-cli@6.2.0
├── bower@1.8.0
├── browserify@12.0.1
├── csslint@1.0.5
├── eslint@4.17.0
├── grunt@0.4.5
├── grunt-cli@1.2.0
├── gulp@3.9.1
├── htmlhint@0.9.13
├── htmllint-cli@0.0.6
├── jshint@2.9.5
├── node@0.0.0
├── nodemon@1.2.1
├── npm@5.6.0
├── sass-lint@1.12.1
└── svgo@1.0.4
```

**Comment by ilyavolodin** [comment]
As I mentioned before, nothing is going to happen if I insert this JSON-LD into HTML file, since ESLint doesn't lint HTML files at all (without plugins. And even with plugin it requires an extra command line flag to even see files with extensions other then ".js"). So there must be something else going on. You can try running ESLint with `--print-config` flag and `--debug` flag to try to figure out where those things configurations are coming from.

**Comment by jaepage** [comment]
Would it not be simple to try the JSON-LD snippet to see what effect you find? 

Running `eslint` with `--config` wants a path to an additional config. If I give it the path to my only config, its output doesn't differ from running `eslint` without the `--config` option, which is: 

```
1:2  error  Parsing error: Unexpected token !

✖ 1 problem (1 error, 0 warnings)
```

If I run it with `--debug`: 
```
eslint:cli Running on files +0ms
eslint:glob-util Creating list of files to process. +0ms
eslint:ignored-paths Looking for ignore file in [path/to/repo] +0ms
eslint:ignored-paths Could not find ignore file in cwd +1ms
eslint:cli-engine Processing [path/to/repo]/index.html +0ms
eslint:cli-engine Linting [path/to/repo]/index.html +1ms
eslint:config Constructing config file hierarchy for [path/to/repo] +0ms
eslint:config Using .eslintrc and package.json files +1ms
eslint:config Using personal config file +2ms
eslint:config Using personal config +0ms
eslint:config-file Loading config file: /Users/JP/.eslintrc +0ms
eslint:config-ops Using config from partial cache +0ms
eslint:config-ops Apply environment settings to config +1ms
eslint:linter Linting code for [path/to/repo]/index.html (pass 1) +0ms
eslint:linter Generating fixed text for [path/to/repo]/index.html (pass 1) +9ms
eslint:text-fixer Applying fixes +0ms
eslint:text-fixer shouldFix parameter was false, not attempting fixes +1ms
eslint:cli-engine Linting complete in: 251ms +242ms
```

**Comment by ilyavolodin** [comment]
So the debug log shows that you also have a config file in your user's directory, that's most likely where plugins are coming from. As I mention, I can't try that JSON-LD snippet, because ESLint doesn't support linting JSON at all. ESLint (as the name implies), only works on ECMAScript out of the box. You could get it to work on other file formats, but that requires installation and configuration of plugins. Since I have no idea which plugins you are using, it's not something I can try.

**Comment by jaepage** [comment]
I don't also have a config file in my user directory. I only have a config file in my user directory, the contents of which I previously provided above. 

Trying the snippet would take less effort than insisting it does nothing.

**Comment by not-an-aardvark** [comment]
Can you be more precise about the specific command that you're using to run ESLint (with all the flags and arguments)?

**Comment by jaepage** [comment]
I ordinarily don't run any command, I let ESLint run in Sublime, and I observe the gutter lights and toolbar messages in the Sublime app. 

When I was requested above to run ESLint in the command line: 

```
$ eslint index.html
[path/to/repo]/index.html
1:2  error  Parsing error: Unexpected token !
✖ 1 problem (1 error, 0 warnings)

$ eslint index.html --print-config --debug
Value for 'print-config' of type 'path::String' required.

$ eslint index.html --debug
eslint:cli Running on files +0ms
eslint:glob-util Creating list of files to process. +0ms
eslint:ignored-paths Looking for ignore file in [path/to/repo] +0ms
eslint:ignored-paths Could not find ignore file in cwd +0ms
eslint:cli-engine Processing [path/to/repo]/index.html +0ms
eslint:cli-engine Linting [path/to/repo]/index.html +0ms
eslint:config Constructing config file hierarchy for [path/to/repo] +0ms
eslint:config Using .eslintrc and package.json files +0ms
eslint:config Using personal config file +2ms
eslint:config Using personal config +0ms
eslint:config-file Loading config file: [~].eslintrc +0ms
eslint:config-ops Using config from partial cache +0ms
eslint:config-ops Apply environment settings to config +1ms
eslint:linter Linting code for [path/to/repo]/index.html (pass 1) +0ms
eslint:linter Generating fixed text for [path/to/repo]/index.html (pass 1) +8ms
eslint:text-fixer Applying fixes +0ms
eslint:text-fixer shouldFix parameter was false, not attempting fixes +0ms
eslint:cli-engine Linting complete in: 233ms +226ms
[path/to/repo]/index.html
1:2  error  Parsing error: Unexpected token !
✖ 1 problem (1 error, 0 warnings)
```

**Comment by not-an-aardvark** [comment]
Thanks, that's useful. To clarify @ilyavolodin's request above, the `--print-config` flag should be used with a filepath argument, e.g. `eslint --print-config index.html`. Can you paste the result of that command?

**Comment by jaepage** [comment]
Thank you for clarifying. I thought `--print-config` was asking for a name of a config file. Giving it the name of the file to lint produces the contents of my config file, same as pasted above. 

```
$ eslint --print-config index.html
{
  "globals": {},
  "env": {},
  "rules": {},
  "parserOptions": {
    "ecmaVersion": 6,
    "sourceType": "module",
    "ecmaFeatures": {
      "jsx": true
    }
  }
}
```

**Comment by kaicataldo** [comment]
Has this been resolved? It doesn't look like there's been any activity in a while.

**Comment by ilyavolodin** [comment]
I think we should close this issue. I'm still not sure what exactly was the problem.
